packages/ui-kit/src/components/mermaid-client-side.tsx

```diff
@@ -111,20 +115,25 @@ const Figure = styled.figure`
 
 const idForGraph = (graph: string) => `mermaid-${murmurhash.v3(graph)}`;
 
-mermaid.initialize(config);
-
 type MermaidProps = {
   graph: string;
 };
 
 const Mermaid = ({ graph }: MermaidProps) => {
   const [svg, setSvg] = useState('');
-
+  const mermaidLoadStatus = useScript(
+    `https://cdn.jsdelivr.net/npm/mermaid@${mermaidVersion}/dist/mermaid.min.js`
+  );
   useEffect(() => {
-    mermaid.mermaidAPI.render(idForGraph(graph), graph, (svg: string) => {
-      setSvg(svg);
-    });
-  }, [graph]);
+    if (mermaidLoadStatus === 'ready') {
+      // eslint-disable-next-line  @typescript-eslint/no-explicit-any
+      const mermaid = (window as any).mermaid;
+      mermaid.initialize(config);
```

Comment:
maybe there is optimization potential in trying to avoid re-initialization of mermaid in every render?